tests/v2/validation/fleet/provisioning/airgap/airgap.go

```diff
@@ -0,0 +1,180 @@
+package airgap
+
+import (
+	"crypto/x509"
+	"errors"
+	"os"
+	"strings"
+
+	"encoding/pem"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	"golang.org/x/crypto/ssh"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	corralBastionIP  = "bastion_ip"
+	corralSSHUser    = "aws_ssh_user"
+	corralPrivateKey = "corral_private_key"
+	systemRegistry   = "system-default-registry"
+
+	fleetExampleFolderName = "fleet-examples.git"
+	rancherGithubURL       = "https://github.com/rancher/"
+	fleetDefaultNS         = "fleet-default"
+
+	sshPort = ":22"
+	rsa     = "RSA"
+	tcp     = "tcp"
+)
+
+func setupAirgapFleetResources(bastionUser, bastionIP, privateKey string) (string, error) {
+	var key ssh.Signer
+	var err error
+
+	if strings.Contains(privateKey, rsa) {
+
+		keyBlock, _ := pem.Decode([]byte(privateKey))
+		if keyBlock == nil {
+			return "", errors.New("unable to use privateKey")
+		}
+
+		rsaKey, err := x509.ParsePKCS1PrivateKey(keyBlock.Bytes)
+		if err != nil {
+			return "", err
+		}
+
+		key, err = ssh.NewSignerFromKey(rsaKey)
+	} else {
+		key, err = ssh.ParsePrivateKey([]byte(privateKey))
+	}
+
+	if err != nil {
+		return "", err
+	}
+
+	return updateLocalFleetRepo(bastionUser, bastionIP, key)
+
+}
+
+// updateLocalFleetRepo clones fleet-examples repo onto the bastion node via ssh
+func updateLocalFleetRepo(bastionUser, bastionIP string, privateKey ssh.Signer) (string, error) {
+	config := &ssh.ClientConfig{
+		User: bastionUser,
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(privateKey),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+
+	conn, err := ssh.Dial(tcp, bastionIP+sshPort, config)
+	if err != nil {
+		return "", err
+	}
+	defer conn.Close()
+
+	// install git if it isn't on the bastion already
+	err = executeCommand(conn, "git version")
+	if err != nil {
+		gitSession, err := conn.NewSession()
+		if err != nil {
+			return "", err
+		}
+		defer gitSession.Close()
+
+		osOutput, err := gitSession.CombinedOutput("hostnamectl")
+		if err != nil {
+			return "", err
+		}
+		osString := string(osOutput)
+
+		if strings.Contains(osString, "Ubuntu") {
+			err = executeCommand(conn, "sudo apt-get update && sudo apt-get install -y git")
+		} else if strings.Contains(osString, "SUSE") {
+			err = executeCommand(conn, "sudo zypper refresh && sudo zypper install git -y")
+		}
+		if err != nil {
+			return "", err
+		}
+	}
+
+	err = executeCommand(conn, "cd ~/ && git clone --mirror "+rancherGithubURL+fleetExampleFolderName)
+	if err != nil {
+		if !strings.Contains(err.Error(), "128") {
+			return "", err
+		}
+	}
+
+	err = executeCommand(conn, "sudo git config --system --add safe.directory '*'")
+	if err != nil {
+		return "", err
+	}
+
+	session, err := conn.NewSession()
```

Comment:
qq, Since executeCommand creates a new session each time, do we want the additional session with conn here, or is it required for the IP output using CombinedOutput?

Reply:
its required for the IP output